go/tasks/pluginmachinery/ioutils/output_sandbox.go

```diff
@@ -0,0 +1,66 @@
+package ioutils
+
+import (
+	"context"
+	"crypto/sha1" // #nosec
+	"encoding/hex"
+
+	"github.com/lyft/flytestdlib/storage"
+
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/io"
+)
+
+type precomputedOutputSandbox struct {
+	path storage.DataReference
+}
+
+func (r precomputedOutputSandbox) GetOutputDataSandboxPath() storage.DataReference {
+	return r.path
+}
+
+// Creates a deterministic OutputSandbox whose path is distributed based on the ShardSelector passed in.
+// Determinism depends on the outputMetadataPath
+// Potential performance problem, as creating anew randomprefixShardedOutput Sandbox may be expensive as it hashes the outputMetadataPath
+// the final OutputSandbox is created in the shard selected by the sharder at the basePath and then appended by a hashed value of the outputMetadata
+func NewRandomPrefixShardedOutputSandbox(ctx context.Context, sharder ShardSelector, basePath, outputMetadataPath storage.DataReference, store storage.ReferenceConstructor) (io.OutputDataSandbox, error) {
```

Comment:
This is not random prefix, right? this depends on the ShardSelector. This guy is pretty deterministic otherwise... right?

Reply:
it is very deterministic by design, i should rename this constructor. Let me think about the names